src/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -406,36 +412,36 @@ public void DiagnosticSourceStartStop()
                     var activity = new Activity("activity");
 
                     // Test Activity.Start
-                    Stopwatch sw = Stopwatch.StartNew();
                     source.StartActivity(activity, arguments);
 
-                    // DateTime.UtcNow is not precise on some platforms, we check that StartTime of Activity is within 20ms from now
-                    Assert.True(Math.Abs((DateTime.UtcNow - observer.Activity.StartTimeUtc).TotalMilliseconds) <= 20);
+                    // We 'fix' DateTime on netfx to be precise; 
+                    // comparing Activity StartTime to potentially imprecise DateTime.UtcNow is not correct
+                    // this test does not intend fo check StartTime/Duration precision, so we allow anything within 20ms.
+                    var acceptableNow = DateTime.UtcNow.AddMilliseconds(20);
```

Comment:
Same feedback applied here.   e.g 

NowTakeBeforeStart -  secMaxError  <= StartTimeUtc
StartTimeUtc < Now + MaxError

THe pattern is that you should only be adding (or subtracting MaxError from Now (or some Now taken at some other point in time).